retroshare-gui/src/gui/chat/ChatWidget.cpp

```diff
@@ -379,31 +379,42 @@ void ChatWidget::init(const ChatId &chat_id, const QString &title)
 
 		if (messageCount > 0)
 		{
-            rsHistory->getMessages(chatId, historyMsgs, messageCount);
+			rsHistory->getMessages(chatId, historyMsgs, messageCount);
 
 			std::list<HistoryMsg>::iterator historyIt;
 			for (historyIt = historyMsgs.begin(); historyIt != historyMsgs.end(); ++historyIt)
-            {
-                // it can happen that a message is first added to the message history
-                // and later the gui receives the message through notify
-                // avoid this by not adding history entries if their age is < 2secs
-                if ((time(NULL)-2) <= historyIt->recvTime)
-                    continue;
-
-                QString name;
-                if (chatId.isLobbyId() || chatId.isDistantChatId())
-                {
-                    RsIdentityDetails details;
-                    if (rsIdentity->getIdDetails(RsGxsId(historyIt->peerName), details))
-                        name = QString::fromUtf8(details.mNickname.c_str());
-                    else
-                        name = QString::fromUtf8(historyIt->peerName.c_str());
-                } else {
-                    name = QString::fromUtf8(historyIt->peerName.c_str());
-                }
+			{
+				// it can happen that a message is first added to the message history
+				// and later the gui receives the message through notify
+				// avoid this by not adding history entries if their age is < 2secs
+				if ((time(nullptr)-2) <= historyIt->recvTime)
```

Comment:
time(NULL) - 2 is a bit dangerous because you're substracting to an unsigned int, so it will be converted to int. Generally, prefer    if(time(nullptr) <= 2+historyIt->recvTime)

Reply:
Done, even if not mine. ;)

ef5fca5e5e (AsamK                        2015-08-07 19:08:10 +0200)                 if ((time(NULL)-2) <= historyIt->recvTime)